src/main/java/com/endercrest/voidspawn/MoveListener.java

```diff
@@ -14,8 +15,13 @@
     @EventHandler
     public void onMoveEvent(PlayerMoveEvent event){
         Player player = event.getPlayer();
-        if(player.getLocation().getBlock().getRelative(BlockFace.DOWN).getType().isSolid()){
+        if(player.getLocation().getBlock().getRelative(BlockFace.DOWN).getType().isSolid() && !isTrapDoor(player.getLocation().getBlock().getRelative(BlockFace.DOWN).getType())){
             TeleportManager.getInstance().setPlayerLocation(player.getUniqueId(), player.getLocation());
         }
     }
-}
\ No newline at end of file
+    
+    public boolean isTrapDoor(Material mat) {
+        return (mat == Material.OAK_TRAPDOOR || mat == Material.SPRUCE_TRAPDOOR || mat == Material.BIRCH_TRAPDOOR || mat == Material.JUNGLE_TRAPDOOR || mat == Material.ACACIA_TRAPDOOR || mat == Material.DARK_OAK_TRAPDOOR || mat == Material.IRON_TRAPDOOR);
```

Comment:
Just to future proof things, could we change this to be using the Tag system? If you apply this suggestion, you will need to add the import as well.
```suggestion
        return Tag.TRAPDOORS.isTagged(mat);
```